var/spack/repos/builtin/packages/py-py-tes/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyPyTes(PythonPackage):
+    """Library for communicating with the GA4GH Task Execution API."""
+
+    pypi = "py-tes/py-tes-0.4.2.tar.gz"
+
+    license("MIT")
+
+    version("0.4.2", sha256="f6926cd59b7dfc8e37840955bf1cc7c43ad4d99ba5eae100b6156c918617472c")
+
```

Comment:
```python
depends_on("python@2.7:3", type=("build", "run"))
```

Reply:
I think you are suggesting this only because the original package states
```
python_requires=">=2.7, <4",
```
but adding this would be inconsistent with other packages and Spack philosophy. That is constraints are only added when we know that something doesn't work. We don't know if it doesn't work with Python 4, so we shouldn't put it. Not to mention that considering Python 4 is unrealistic.

By the way, does Spack still support Python 2?

Reply:
Spack supports Python 3.8+, I'm not actually concerned about the lower bound.

We don't know if it will work with Python 4, but the install process will explicitly fail if you try because the authors are using `python_requires`. I agree it's silly to consider Python 4, although it never hurts to be futureproof.

This is an optional request. If you'd rather not specify and risk this package breaking in the unlikely event Python 4 is ever released, I'm fine with merging as is.

Reply:
> but the install process will explicitly fail if you try because the authors are using `python_requires`

This makes sense, I have added the requested line.